Fix a deadlock with empty settings (#5546)
koreader/koreader: doc/Building.md

./kodev run
```

> *Note:* On non-Linux hosts, you might want to pass `--no-build` to prevent re-running the buildsystem, as incremental builds are probably not behaving properly.

Reviewer: How do you mean?

Reviewer: I wouldn't appreciate that. ^_^ (Even on my slow laptop it's less than half a second's delay when nothing needs to be done.)

Anyway, surely that just means the Mac build somehow depends on the wrong files? Weird though.

Author: No idea, so I figured a warning in the docs would be good enough ;).

Reviewer: Since it seems to be Mac/Xcode that broke itself and afaik not *BSD, Windows, etc. ;-)
```suggestion
*Note:* On Mac and possibly other non-Linux hosts, you might want to pass `--no-build` to prevent re-running the buildsystem, as incremental builds may not behave properly.
```

Reviewer: Occasionally something might get slightly confused when a dependency changes; does it still happen if you clear out the thirdparty/lua-spore/build/x86_64-etc subfolder? (Or just the whole build one if you're feeling lazy but then it'll needlessly redownload.)

Reviewer 2: yes, i narrowed it down.

Cmake expects:
build/x86_64-pc-linux-gnu-debug/rocks/lib/luarocks/**rocks**/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec

my luarocks (luarocks 3.2.1) creates:
/AUR/koreader/base/build/x86_64-pc-linux-gnu-debug/rocks/lib/luarocks/**rocks-5.1**/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec

this is due to luarocks 3.x supporting one luarocks with multiple lua versions.
debian and friends still use 2.x

when i change https://github.com/koreader/koreader-base/blob/2700c5315d0307221a0c49c8bbb2508bdf9432f0/Makefile.defs#L799
`LUAROCKS_DIR=$(OUTPUT_DIR)/rocks/lib/luarocks/rocks`
to 
`LUAROCKS_DIR=$(OUTPUT_DIR)/rocks/lib/luarocks/rocks-5.1`
it no longer rebuilds.

is there a way to fix this, use `rocks-5.1` if luarocks is >= 3.x, i cant find a way to change luarocks behavior.
should i open an issue in base?

Reviewer 2: see https://github.com/luarocks/luarocks/blob/master/CHANGELOG.md#user-visible-changes for more info

Reviewer: > should i open an issue in base?

Yes please. Some random comment in an unrelated issue is hard to find.

Reviewer 2: see koreader/koreader-base#1003

Reviewer: That's not an issue. :-P
